spec/policy_machine_storage_adapters/active_record_spec.rb

```diff
@@ -400,9 +401,44 @@
     end
 
     describe '#descendants' do
-      # TODO normalize return value types
-      it 'returns appropriate descendants' do
-        expect(@u1.descendants).to match_array @user_attributes.map(&:stored_pe)
+      context 'no filter is applied' do
+        # TODO normalize return value types
+        it 'returns appropriate descendants' do
+          expect(@u1.descendants).to match_array @user_attributes.map(&:stored_pe)
+        end
+      end
+
+      context 'a filter is applied' do
+        before do
+          @ua1.update(color: 'green')
+
+          @new_ua = @pm.create_user_attribute('new_ua')
+          @new_ua.update(color: 'green')
+          @pm.add_assignment(@u1, @new_ua)
+        end
+
+        it 'applies a single filter if one is supplied' do
+          green_descendants = @u1.descendants(color: 'green')
+          expect(green_descendants.size).to eq 2
+          expect(green_descendants).to contain_exactly(@ua1.stored_pe, @new_ua.stored_pe)
+        end
+
+        it 'applies multiple filters if they are supplied' do
+          green_descendants = @u1.descendants(color: 'green', unique_identifier: 'new_ua')
+          expect(green_descendants.size).to eq 1
+          expect(green_descendants).to contain_exactly(@new_ua.stored_pe)
+        end
+
+        it 'returns appropriate results when filters apply to no descendants' do
+          expect(@u1.descendants(color: 'taupe')).to be_empty
+          expect { @u1.descendants(not_a_real_attribute: 'fake') }.to raise_error(ArgumentError)
+        end
+
+        it 'returns appropriate results when filters apply to all descendants' do
+          all_descendants = @user_attributes.map(&:stored_pe) + [@new_ua.stored_pe]
+          expect(@u1.descendants({})).to match_array(all_descendants)
```

Comment:
`contain_exactly` takes a comma-separated list of elements, `match_array` takes an array. It's incredibly minor, just depends on what data you have lying around.